### DISCUSSION TITLE
请教哪一个模型可以识别竖排排版的文字，比如老的古文和一些古籍

### DISCUSSION THREAD

**Discussion by fattank** [PaddlePaddle/PaddleOCR#12330] [Q&A]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：Win10
- 版本号/Version：Paddle：  PaddleOCR：2.7 问题相关组件/Related components：
- 运行指令/Command Code：
- 完整报错/Complete Error Message：

请尽量不要包含图片在问题中/Please try to not include the image in the issue.
请教哪一个模型可以识别竖排，排版的文字，
![612f39d95df6e](https://github.com/PaddlePaddle/PaddleOCR/assets/23254194/3147f2d0-a298-4bf9-9c8a-1a50f0dc206d)
比如老的古文和一些古籍

**Comment by tink2123** [comment]
ppocr系列的文本检测模型都可以～ 如果效果不满足效果，建议使用自己的数据微调

**Comment by lhw828** [comment]
竖排可以识别，但是保存的时候，版式有问题。

**Reply by PandaBalls** [reply]
我遇到的也是版式问题，识别文字单个都很准确，但是返回的字符块全都是横排的行。语义都是不对的。不知道如何强制返回列识别

**Comment by HungryFlo** [comment]
我想微调ppocrv4文本检测模型det和文本识别模型rec来做古籍检测，但是官方提供的文本识别模型微调参数中如果修改sampler的参数将其改成竖条模型就会报错维度不匹配，这种情况应该如何微调呢？
下面是官方给出的ch_PP-OCRv4_rec.yml中的部分
> sampler:
>     name: MultiScaleSampler
>     scales: [[320, 32], [320, 48], [320, 64]]
>     first_bs: &bs 192
>     fix_bs: false
>     divided_factor: [8, 16] # w, h
>     is_training: True
感谢🙏

**Reply by liuhongen1234567** [reply]
您好，这种竖版文本 ppocrv4 在训练时会将其旋转成水平文本进行训练。所以不需要修改维度
<img width="764" height="168" alt="image" src="https://github.com/user-attachments/assets/904033e7-9039-40c9-a4e7-4dc308b0b224" />
